### PR TITLE
DRY the build

### PR DESCRIPTION
**Pull request by NeQuissimus**
Bring back JDK11 into build matrix

### CODE REVIEW

**Comment by softinio** [comment]
This work is awesome! Very impressed. 

Do we have a use case to use CircleCI orbs?

**Comment by NeQuissimus** [comment]
Don't be impressed, nothing works (yet :D).

I did not see much value in orbs.

**Comment by NeQuissimus** [comment]
See #1022 for why the Dotty failures are happening.
Obviously, I cannot test the release steps.
Otherwise, I think this is good to go.

**Comment by NeQuissimus** [comment]
This closes #979

**Comment by softinio** [comment]
@NeQuissimus Noticed you are doing a release to test this. Last release 2.11 artifacts didn't get published. Took a closer look and it fails but doesn't stop build continuing. Same seems to happening with your build: 

https://circleci.com/gh/zio/zio/1725

from the 2.11 release step: 

```
[error] /root/project/core/jvm/src/test/scala/zio/IOSpec.scala:180:33: File name too long
[error] This can happen on some encrypted or legacy file systems.  Please see SI-3623 for more details.
[error]         (conditionVal2 must_=== 2) and
[error]                
```

```
[error] one error found
```

```
[error] (coreJVM / Test / compileIncremental) Compilation failed
[error] Total time: 100 s, completed Jun 16, 2019 6:54:45 PM
```

But still shows a success at end of build. So lets see if the publish failed again and its due to these or not.

cc @mijicd

**Comment by NeQuissimus** [comment]
Well, that's pretty wacky...

**Comment by iravid** [comment]
Just move all the steps to use machine executors. That’s an issue with Docker’s overlay filesystem.

**Comment by NeQuissimus** [comment]
Let me disable the dotty builds tomorrow and then we should be able to merge this

**Comment by softinio** [comment]
Good idea. Once merged let me know what's outstanding and I will try to help. Didn't want to step on your toes whilst you were working on this awesome PR.

**Comment by NeQuissimus** [comment]
Here we go

**Comment by jdegoes** [comment]
@NeQuissimus If we weren't running Dotty tests before, I think we can skip that for now; that way we won't be regressing (smaller chunk to bite off at once).

What's the status on this one?

**Comment by iravid** [comment]
We weren't running the Dotty tests before because of some incompatibilities between specs2 and Dotty

**Comment by NeQuissimus** [comment]
Everything here is done. The dotty tests need to be enabled later.

**Comment by jdegoes** [comment]
@NeQuissimus Do you know why there are old Circle CI status that hang forever on a lot of pull requests?

**Comment by iravid** [comment]
@jdegoes That's because the build has been refactored and the step names were changed, but the repo settings are still pointing to the previous step names. Once this PR is merged, we can change the settings.

Older PRs would still need a rebase/merge though.

**Comment by jdegoes** [comment]
@iravid Looks to me like Circle CI passed. Any objections to merging this now?

**Comment by iravid** [comment]
Nope! Merging :-)

**Comment by iravid** [comment]
@NeQuissimus can you change the required status checks settings?

**Comment by softinio** [comment]
👏

**Comment by NeQuissimus** [comment]
@iravid looks like it's already done

**Comment by jdegoes** [comment]
@NeQuissimus I tries to do it, but I wasn't 100% sure which to enable or disable...

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,175 +1,142 @@
...
+testJVM: &testJVM
+  steps:
+    - checkout
+    - <<: *load_cache
+    - <<: *install_jdk
+    - run:
+        name: Run tests
+        command: ./sbt ++${SCALA_VERSION}! testJVM
+    - run:
+        name: Package artifacts
```
Comment (ghostdogpr) [inline .circleci/config.yml]: Could we do this only when we release? It takes 3-4 min for every test step.

Comment (NeQuissimus) [reply .circleci/config.yml]: I don't see why not, just rip this `run` out. The release steps will package anyways if it has not happened already

File: .circleci/config.yml [hunk 2]
```diff
@@ -180,91 +147,187 @@ jobs:
...
+      - <<: *jdk_8
 
 workflows:
   version: 2
   build:
     jobs:
-      - lint212_jdk8:
-          filters:
-            tags:
-              only: /^v(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(?:-((?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?$/
```
Comment (ghostdogpr) [inline .circleci/config.yml]: @NeQuissimus any reason to remove this? By default, builds are not triggered on tags : https://circleci.com/docs/2.0/workflows/#executing-workflows-for-a-git-tag
I just tried to cut a release and nothing happened indeed.

Comment (NeQuissimus) [reply .circleci/config.yml]: Oh crap, I had meant to extract the regex' into a reference but forgot to out it back 😔

**Pull request merged**